Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -456,7 +456,50 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
         ],
         handledNodes: [argument.id]
       )
-      return .visitChildren
+    }
+    if node.hasError,
+       let unexpectedBeforeAtSignTokens = node.unexpectedBeforeAtSignToken,
+       let unexpectedBeforeAtSign = unexpectedBeforeAtSignTokens.lastToken(viewMode: .sourceAccurate) {
```

Comment:
Wit this implementation you would be discarding any tokens before the `@` sign, but the diagnostic message only complains about whitespace. What I would do instead is to check that `unexpectedBeforeAtSignTokens` only contains a single `@` token as follows:

```swift
let unexpectedBeforeAtSign = unexpectedBeforeAtSignTokens.onlyToken(where: { $0.tokenKind == .atSign && !$0.trailingTrivia.isEmpty })
```